scripts/autotools/libmongocrypt/CheckSSL.m4

```diff
@@ -0,0 +1,35 @@
+if test "$PHP_CLIENT_SIDE_ENCRYPTION" != "no"; then
+  AC_MSG_CHECKING([which crypto library to use for libmongocrypt])
+  AC_MSG_RESULT([$PHP_MONGODB_SSL])
+
+  dnl Disable Windows crypto
+  AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_CNG, 0)
+
+  if test "$PHP_MONGODB_SSL" = "darwin"; then
+    PHP_CLIENT_SIDE_ENCRYPTION="yes"
+
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_LIBCRYPTO, 0)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_COMMON_CRYPTO, 1)
+
+    PHP_MONGODB_LIBMONGOCRYPT_CFLAGS="$PHP_MONGODB_LIBMONGOCRYPT_CFLAGS -DKMS_MESSAGE_ENABLE_CRYPTO=1 -DKMS_MESSAGE_ENABLE_CRYPTO_COMMON_CRYPTO=1"
+  elif test "$PHP_MONGODB_SSL" = "openssl" -o "$PHP_MONGODB_SSL" = "libressl"; then
+    PHP_CLIENT_SIDE_ENCRYPTION="yes"
+
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_LIBCRYPTO, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_COMMON_CRYPTO, 0)
+
+    PHP_MONGODB_LIBMONGOCRYPT_CFLAGS="$PHP_MONGODB_LIBMONGOCRYPT_CFLAGS -DKMS_MESSAGE_ENABLE_CRYPTO=1 -DKMS_MESSAGE_ENABLE_CRYPTO_LIBCRYPTO=1"
+  elif test "$PHP_CLIENT_SIDE_ENCRYPTION" = "auto"; then
+    PHP_CLIENT_SIDE_ENCRYPTION="no"
+
+    AC_MSG_RESULT(No SSL library found. Compiling without libmongocrypt. Please specify a library using the --with-mongodb-ssl option)
+    AC_SUBST(MONGOC_ENABLE_CLIENT_SIDE_ENCRYPTION, 0)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO, 0)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_LIBCRYPTO, 0)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_COMMON_CRYPTO, 0)
```

Comment:
I also think we can remove these `AC_SUBST` calls, as `PHP_CLIENT_SIDE_ENCRYPTION` being "no" means libmongocrypt will be entirely disabled. Once we get back to `config.m4`, we're not going to bother adding its sources, nor will we make other `AC_SUBST` calls that might otherwise be required for a bundled build (e.g. `AC_SUBST(MONGOCRYPT_IS_POSIX, 1)`).